build/platforms.bzl

```diff
@@ -85,24 +102,37 @@ def _update_dict_for_platform_category(d, value, platforms, only_os = None):
             continue
         for arch in arches:
             constraint = go_platform_constraint(os, arch)
-            if type(value) == "list":
+            fmt_args = {"OS": os, "ARCH": arch}
+            if types.is_list(value):
+                # Format all items in the list, and hope there are no duplicates
                 d.setdefault(constraint, []).extend(
-                    [v.format(OS = os, ARCH = arch) for v in value],
+                    [v.format(**fmt_args) for v in value],
                 )
             else:
+                # Don't overwrite existing value
                 if constraint in d:
                     fail("duplicate entry for constraint %s", constraint)
-                d[constraint] = value.format(OS = os, ARCH = arch)
+                if types.is_dict(value):
+                    # Format dictionary values only
+                    d[constraint] = {
+                        dict_key: dict_value.format(**fmt_args)
+                        for dict_key, dict_value in value.items()
+                    }
+                else:
+                    # Hopefully this is just a string
+                    d[constraint] = value.format(**fmt_args)
 
 def for_platforms(
```

Comment:
Can we get some documentation on how this works?

Also if this is the only public method in the file, does it make more sense as the first method?

Reply:
the documentation is now significantly longer than the implementation.